src/plugins/data/public/search/search_service.ts

```diff
@@ -149,6 +149,15 @@ export class SearchService implements Plugin<ISearchSetup, ISearchStart> {
     };
   }
 
+  // Get language id from the request if available. QueryStringManager returns
+  // the global language id, which won't work in dashboard with different
+  // visualizations.
+  private getLanguageId(request: Parameters<ISearchStart['search']>[0]): string | undefined {
+    if (request.params?.body?.query?.queries?.[0]?.language)
+      return request.params.body.query.queries[0].language;
```

Comment:
i think this might have an issue. can you verify.

when the user types in a query and is enhancements is enabled then firing a query on a dashboard should be queries[1] where query[0] is the viz query.

my understanding of queries is like a stack. you want the top (highest index) query be used

Reply:
checked with @kavilla offline, the `queries` for PPL always contains 1 element regardless of user query in search bar. https://github.com/opensearch-project/OpenSearch-Dashboards/blob/0daad39467004cd06bd784ec154bcbb1b2e6ea63/src/plugins/data/common/opensearch_query/opensearch_query/build_opensearch_query.ts#L54-L78

If we add PPL support in the search bar, then this may change, but so far there is no plan

For DSL visualizations, `queries` is not used because it returns the built DSL query directly